bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/BsonDecoder.kt

```diff
@@ -45,34 +50,93 @@ import org.bson.types.ObjectId
  *
  * For custom serialization handlers
  */
-public sealed interface BsonDecoder {
+@ExperimentalSerializationApi
+internal sealed interface BsonDecoder : Decoder, CompositeDecoder {
```

Comment:
As this is sealed there is no need for it to have been public

Reply:
I'll revert as its not strictly true - users could in theory use this aside from the Codecs.